serde/src/de/mod.rs

```diff
@@ -245,211 +307,147 @@ pub trait Deserializer {
 
     /// This method hints that the `Deserialize` type is expecting a `f32` value.
+    /// A reasonable default is to forward to `deserialize_f64` and cast the yielded value.
     #[inline]
     fn deserialize_f32<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize_f64(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting a `f64` value.
     #[inline]
     fn deserialize_f64<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting a `char` value.
     #[inline]
     fn deserialize_char<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting a `&str` value.
     #[inline]
     fn deserialize_str<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting a `String` value.
     #[inline]
     fn deserialize_string<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize_str(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting an `unit` value.
     #[inline]
     fn deserialize_unit<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting an `Option` value. This allows
     /// deserializers that encode an optional value as a nullable value to convert the null value
     /// into a `None`, and a regular value as `Some(value)`.
     #[inline]
     fn deserialize_option<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting a sequence value. This allows
     /// deserializers to parse sequences that aren't tagged as sequences.
     #[inline]
     fn deserialize_seq<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting a fixed size array. This allows
     /// deserializers to parse arrays that aren't tagged as arrays.
-    ///
-    /// By default, this deserializes arrays from a sequence.
     #[inline]
     fn deserialize_fixed_size_array<V>(&mut self,
```

Comment:
Let's call this `deserialize_seq_fixed_size`.

In the Serializer we renamed `serialize_fixed_size_array` to `serialize_seq_fixed_size` to make it a little less confusing that you need to use it with `serialize_seq_elt` and `serialize_seq_end` (instead of some `serialize_fixed_size_array_elt` and `serialize_fixed_size_array_end`) and `SeqState` (instead of `FixedSizeArrayState`).

That reasoning does not apply here, but it would be nice for the name to be consistent across Serializer and Deserializer.